src/taipy/core/cycle/_cycle_repository_mixin.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from datetime import datetime
+from typing import Callable, List
+
+from taipy.config.common.frequency import Frequency
+
+from ..cycle.cycle import Cycle
+
+
+class _CycleRepositoryMixin:
+
+    # TODO: Maybe we should move this to CycleManager class
```

Comment:
These methods don't call any repository service. They are just cycle filtering methods. 
Two alternatives for me: 
1. We move them to the manager
2. We do not provide the cycles as a parameter; we call the repository directly to retrieve the cycles. However, this depends on the way the methods are used.
 
I would go for the second alternative 'cause the filtering could be implemented directly in the query for better performances (in particular with SQL or Mongo). But it should be part of another ticket.
What do you think ?

Reply:
I think thats where the `filters` parameter that I've added to the `_load` methods should be usefull. The idea would be to use the base method to filter the entities needed. The only thing I have against option 2 is that we would have to repeat most of the code from these methods in each Cycle repository (both in mongo and sql). With the mixin we could use the multiple inheritance to reuse the code, that's why I put the cycles as parameter, so we should be able to retrieve the cycles with the necessary filters using `_load_all` and pass to these functions.

But in either case, I think we should discuss it a little more and handle it in a separate PR as you suggested.